metagenomics.py

```diff
@@ -1143,6 +1144,162 @@ class KrakenBuildError(Exception):
+        Full data returned via --jsonOut (filtered by --topN and --countThreshold), whereas -csvOut returns a summary.
+    """
+
+    samples = {}
+    same_level = False
+    
+    def indent_len(in_string):
+        return len(in_string)-len(in_string.lstrip())
+
+    for f in list(summary_files_in):
+        sample_name, extension = os.path.splitext(f)
+        sample_summary = {}
+        tax_headings_copy = [s.lower() for s in tax_headings]
+
+        with util.file.open_or_gzopen(f, 'rU') as inf:
+            should_process = False
+            indent_of_selection = -1
+            currently_being_processed = ""
+            for line in inf:
+                if len(line.rstrip('\r\n').strip()) == 0:
+                    continue
+                csv.register_dialect('kraken_report', quoting=csv.QUOTE_MINIMAL, delimiter="\t")
+                row = next(csv.reader([line.strip().rstrip('\n')], dialect="kraken_report"))
+                
+                indent_of_line = indent_len(row[5])
+                row = [item.strip() for item in row] # remove leading/trailing whitespace from each item
+                
+                # rows are formatted like so:
+                # 0.00  16  0   D   10239     Viruses
+                #
+                # row[0] Percentage of reads covered by the clade rooted at this taxon
+                # row[1] Number of reads covered by the clade rooted at this taxon
+                # row[2] Number of reads assigned directly to this taxon
+                # row[3] A rank code, indicating (U)nclassified, (D)omain, (K)ingdom, (P)hylum, (C)lass, (O)rder, (F)amily, (G)enus, or (S)pecies. All other ranks are simply '-'.
+                # row[4] NCBI taxonomy ID
+                # row[5] indented scientific name
+
+                if indent_of_line == indent_of_selection:
+                    should_process = False
+                    indent_of_selection=-1
+
+                if indent_of_selection == -1:
+                    if row[5].lower() in tax_headings_copy:
+                        tax_headings_copy.remove(row[5].lower())
+                        
+                        should_process = True
+                        indent_of_selection = indent_of_line
+                        currently_being_processed = row[5]
+                        sample_summary[row[5]] = collections.OrderedDict()
+                        if row[3] == rank_code(taxlevel_focus):
+                            same_level = True
+                        if row[3] == "-":
+                            log.warning("Non-taxonomic parent level selected")
+                
+                if should_process:
+                    if (rank_code(taxlevel_focus) == row[3]):
+                        
+                        if int(row[1])>=count_threshold:
+                            sample_summary[currently_being_processed][row[5]] = [float(row[0]), int(row[1])]
```

Comment:
Make values NamedTuples here, or just store the entire row in `sample_summary` ?